keps/sig-apps/3998-job-success-completion-policy/README.md

```diff
@@ -0,0 +1,860 @@
+  - Components exposing the metric:
+
+###### Are there any missing metrics that would be useful to have to improve observability of this feature?
+
+No.
+
+### Dependencies
+
+###### Does this feature depend on any specific services running in the cluster?
+
+No.
+
+### Scalability
+
+No.
+
+###### Will enabling / using this feature result in any new API calls?
+
+No.
+
+###### Will enabling / using this feature result in introducing new API types?
+
+No.
+
+###### Will enabling / using this feature result in any new calls to the cloud provider?
+
+No.
+
+###### Will enabling / using this feature result in increasing size or count of the existing API objects?
+
+Yes, it will increase the size of existing API objects only when the `.spec.successPolicy` is set.
+
+- API type(s): Job
+- Estimated increase in size: `.spec.successPolicy.criteria.succeededIndexes` field are impacted.
+In the worst case, the size of `succeededIndexes` can be estimated about 64KiB (see [Risks and Mitigations](#risks-and-mitigations)).
+
+###### Will enabling / using this feature result in increasing time taken by any operations covered by existing SLIs/SLOs?
+
+No.
+
+###### Will enabling / using this feature result in non-negligible increase of resource usage (CPU, RAM, disk, IO, ...) in any components?
+
+No.
+
+###### Can enabling / using this feature result in resource exhaustion of some node resources (PIDs, sockets, inodes, etc.)?
+
+No.
+
+### Troubleshooting
+
+###### How does this feature react if the API server and/or etcd is unavailable?
+
+The job controller will declare that the job is "Succeeded" based on the `.status.completedIndexes`.
+So, in this case, the job controller can not correctly evaluate the successPolicy 
+because the `.status.completedIndexes` won't be updated.
+
+###### What are other known failure modes?
+
+None.
+
+###### What steps should be taken if SLOs are not being met to determine the problem?
+
+If a successPolicy isn't respected even though the job doesn't match other policies 
+such as a [pod failure policy](https://kubernetes.io/docs/tasks/job/pod-failure-policy/) and backoffLimit:
+
+- Check reachability between Kubernetes components.
+- Consider increasing the logging level to trace when the issues occur.
+- Check the job controller's `job_sync_duration_seconds` metric to check if the job controller's processing duration increases. 
+
+If many requests are rejected, re-queued many times or increased the job controller's processing duration,
+consider tuning the parameters for [APF](https://kubernetes.io/docs/concepts/cluster-administration/flow-control/).
+
+## Implementation History
+ 
+- 2023.06.06: This KEP is ready for review.
+- 2023.06.09: API design is updated.
+- 2023.10.03: API design is updated.
+
+## Drawbacks
+
+Adds more complexity to the criteria to be terminated Job.
+
+## Alternatives
+
+### Relax a validation for completions
+Currently, a job is restricted `.spec.completion!=nil`.
```

Comment:
Not quite true, you can create non-indexed job w/o specifying completions, this allows to mark the job as completed when first pod succeeds. So it'd be good to make it clear that this only applies for indexed jobs.